src/bun.js/api/bun/h2_frame_parser.zig

```diff
@@ -0,0 +1,2440 @@
+                remoteSettings.updateWith(unit);
+            }
+            this.readBuffer.reset();
+            this.remoteSettings = remoteSettings;
+            this.dispatch(.onRemoteSettings, remoteSettings.toJS(this.handlers.globalObject));
+            return content.end;
+        }
+        // needs more data
+        return data.len;
+    }
+
+    fn handleReceivedStreamID(this: *H2FrameParser, streamIdentifier: u32) ?*Stream {
+        // connection stream
+        if (streamIdentifier == 0) {
+            return null;
+        }
+
+        // already exists
+        if (this.streams.getEntry(streamIdentifier)) |entry| {
+            return entry.value_ptr;
+        }
+
+        if (streamIdentifier > this.lastStreamID) {
+            this.lastStreamID = streamIdentifier;
+        }
+
+        // new stream open
+        const settings = this.remoteSettings orelse this.localSettings;
+        var entry = this.streams.getOrPut(streamIdentifier) catch @panic("OOM");
+        entry.value_ptr.* = Stream.init(streamIdentifier, settings.initialWindowSize, this);
+
+        this.dispatch(.onStreamStart, JSC.JSValue.jsNumber(streamIdentifier));
+        return entry.value_ptr;
+    }
+
+    pub fn readBytes(this: *H2FrameParser, bytes: []u8) usize {
+        log("read", .{});
+        if (this.currentFrame) |header| {
+            log("current frame {} {} {} {}", .{ header.type, header.length, header.flags, header.streamIdentifier });
+
+            const stream = this.handleReceivedStreamID(header.streamIdentifier);
+            return switch (@as(FrameType, @enumFromInt(header.type))) {
+                FrameType.HTTP_FRAME_SETTINGS => this.handleSettingsFrame(header, bytes),
+                FrameType.HTTP_FRAME_WINDOW_UPDATE => this.handleWindowUpdateFrame(header, bytes, stream),
+                FrameType.HTTP_FRAME_HEADERS => this.handleHeadersFrame(header, bytes, stream),
+                FrameType.HTTP_FRAME_DATA => this.handleDataFrame(header, bytes, stream),
+                FrameType.HTTP_FRAME_CONTINUATION => this.handleContinuationFrame(header, bytes, stream),
+                FrameType.HTTP_FRAME_PRIORITY => this.handlePriorityFrame(header, bytes, stream),
+                FrameType.HTTP_FRAME_PING => this.handlePingFrame(header, bytes, stream),
+                FrameType.HTTP_FRAME_GOAWAY => this.handleGoAwayFrame(header, bytes, stream),
+                FrameType.HTTP_FRAME_RST_STREAM => this.handleRSTStreamFrame(header, bytes, stream),
+                else => {
+                    this.sendGoAway(header.streamIdentifier, ErrorCode.PROTOCOL_ERROR, "Unknown frame type", this.lastStreamID);
+                    return bytes.len;
+                },
+            };
+        }
+
+        // nothing to do
+        if (bytes.len == 0) return bytes.len;
+
+        const buffered_data = this.readBuffer.list.items.len;
+
+        var header: FrameHeader = undefined;
```

Comment:
are you sure this can be undefined and doesn't need to be zero'd?

Reply:
if yes, please leave a comment

Reply:
We are setting the entire value using from, but I will zero it out, sounds better than undefined.